packages/blaze/view.js

```diff
@@ -352,13 +352,16 @@ Blaze._materializeView = function (view, parentView, _workStack, _intoArray) {
       view._isInRender = false;
 
       if (! c.firstRun) {
+        // added by ramezrafla@zegenie to skip affecting DOM for unchanged content
+        if (Blaze._deepCompare(lastHtmljs,htmljs)) {
```

Comment:
Right, we had deep variables. I think for a complex App, it is to be expected. 

Here is your example adapted to our test case:
``
var obj = { a: { b : 1}};
Collection.update({}, {$set: obj});
``